app/controllers/application_controller.rb

```diff
@@ -198,8 +214,7 @@ def redirect_back_or(path)
 
   def redirect_to_siginin_if_anon_or(path)
     if current_user.nil?
-      session[:redirect_path_after_signin] = request.path
-      redirect_to new_user_session_path
+      redirect_to auth_login_path(after_sign_in_path: request.path)
     elsif !path.empty?
```

Comment:
Also if we make that change it would probably make sense to modify the way `params[:redirecting_after_sign_in]` is handled. It should really only be checked if there is a current_user.

```
if current_user
  # comment explaining this...
  flash[:alert] = error_message if not params[:redirecting_after_sign_in]
  redirect_to after_sign_in_path_for(current_user)
else
  flash[:alert] = error_message
  redirect_to auth_login_path(after_sign_in_path: request.path)
end
```